hub/policy/src/main/java/uk/gov/ida/hub/policy/services/CountriesService.java

```diff
@@ -53,8 +54,12 @@ public void setSelectedCountry(SessionId sessionId, String countryCode) {
         Class<? extends State> expectedStateClass = SessionStartedState.class;
         if (sessionRepository.isSessionInState(sessionId, EidasCountrySelectedState.class)) {
             expectedStateClass = EidasCountrySelectedState.class;
+        } else if (sessionRepository.isSessionInState(sessionId, EidasAuthnFailedErrorState.class)) {
+            expectedStateClass = EidasAuthnFailedErrorState.class;
         }
 
+
+
         EidasCountrySelectingStateController eidasCountrySelectingStateController = (EidasCountrySelectingStateController) sessionRepository.getStateController(sessionId, expectedStateClass);
```

Comment:
We can completely remove the `expectedStateClass` variable since we always expect the state to implement `EidasCountrySelectingState`